app/javascript/hyacinth_v1/utils/redirectToSearch.js

```diff
@@ -0,0 +1,19 @@
+import FilterArrayParam from './filterArrayParam';
+import * as qs from 'query-string';
+import {
+  NumberParam, StringParam, withDefault, encodeQueryParams
+} from 'use-query-params';
+
+export const queryParamsConfig = {
```

Comment:
I tested this branch locally and I'm seeing warnings in the js console like `encodeQueryParams.js:18 Encoding parameter totalCount as string since it was not configured.`.  You should be able to see the same thing if you open the js console after doing a search and clicking on a digital object show page.

I think this was a problem with the code before your changes, so nothing to do with your new code, but would you mind updating `queryParamsConfig` in `redirectToSearch.js` so that it covers all of the params being passed in here to `const queryParams`?  That should hopefully get rid of the console warnings.

Reply:
One remaining parameter, 'query', triggers this warning. I need to test it out, but I think this could be resolved if searchParams used the key 'q' rather than 'query'.

Reply:
Sounds good!  Also, the latest commits you added look good.  Feel free to re-request a review from me on this PR when you're finished with the q/query change and I'll check it out.

Reply:
The q/query issue seems to have required only a change in one spot. I think we're done here.

Reply:
👍